service/header/sync.go

```diff
@@ -14,24 +16,35 @@ type Syncer struct {
 	store    Store
 	trusted  tmbytes.HexBytes
 
-	// done is triggered when syncing is finished
-	// it assumes that Syncer is only used once
-	done chan struct{}
+	// inProgress is set to 1 once syncing commences and
+	// is set to 0 once syncing is either finished or
+	// not currently in progress
+	inProgress uint64
 }
 
 // NewSyncer creates a new instance of Syncer.
 func NewSyncer(exchange Exchange, store Store, trusted tmbytes.HexBytes) *Syncer {
 	return &Syncer{
-		exchange: exchange,
-		store:    store,
-		trusted:  trusted,
-		done:     make(chan struct{}),
+		exchange:   exchange,
+		store:      store,
+		trusted:    trusted,
+		inProgress: 0, // syncing is not currently in progress
 	}
 }
 
+// Start starts the syncing routine.
+func (s *Syncer) Start(context.Context) error {
+	go s.Sync(context.TODO()) // TODO @Wondertan: leaving this to you to implement in disconnection toleration PR.
```

Comment:
Is there an issue for that "disconnection toleration PR"?

Reply:
Yes, can't find it from phone now